enumerables.rb

```diff
@@ -0,0 +1,155 @@
+module Enumerable
+    def my_each
+        for i in 0...self.length do
+            yield(self[i])
+        end
+    end
+    
+    def my_each_with_index
+        for i in 0...self.length do
+            yield(self[i], i)
+        end
+    end
+
+    def my_select
+        a = []
+        my_each do |n|
+            if yield(n)
+                a.push(n)
+            end
+        end
+        a
+    end
+
+    def my_all?
+        my_each do |n|
+            if !yield(n)
+                return false
+            end
+        end
+        true
+    end
+
+    def my_none?
+        my_each do |n|
+            if yield(n)
+                return false
+            end
+        end
+        true
+    end
+
+    def my_count(arg = nil)
+            count = 0
+            if arg == nil && block_given? == false
+            return self.length    
+            elsif block_given?
+                my_each do |n|
+                    if yield(n)
+                    count += 1
+                    end
+                end
+                return count
+            else
+            my_each do |n|
+                if arg == n
+                count += 1
+                end
+            end
+            return count
+        end  
+    end
+
```

Comment:
- [ ] my_count should return the number of items in enum through enumeration::range, and counts the number of items in enum that are equal to item if an argument is given